compute/cloud-client/src/main/java/compute/custommachinetype/CreateWithHelper.java

```diff
@@ -0,0 +1,287 @@
+    static {
+      Map<String, CpuSeries> map = new ConcurrentHashMap<>();
+      for (CpuSeries instance : CpuSeries.values()) {
+        map.put(instance.getCpuSeries(), instance);
+      }
+      ENUM_MAP = Collections.unmodifiableMap(map);
+    }
+
+    private final String cpuSeries;
+
+    CpuSeries(String cpuSeries) {
+      this.cpuSeries = cpuSeries;
+    }
+
+    public static CpuSeries get(String name) {
+      return ENUM_MAP.get(name);
+    }
+
+    public String getCpuSeries() {
+      return this.cpuSeries;
+    }
+  }
+
+  static final class TypeLimits {
+
+    int[] allowedCores;
+    int minMemPerCore;
+    int maxMemPerCore;
+    int extraMemoryLimit;
+    boolean allowExtraMemory;
+
+    TypeLimits(int[] allowedCores, int minMemPerCore, int maxMemPerCore, boolean allowExtraMemory,
+        int extraMemoryLimit) {
+      this.allowedCores = allowedCores;
+      this.minMemPerCore = minMemPerCore;
+      this.maxMemPerCore = maxMemPerCore;
+      this.allowExtraMemory = allowExtraMemory;
+      this.extraMemoryLimit = extraMemoryLimit;
+    }
+  }
+
+  // The limits for various CPU types are described on:
+  // https://cloud.google.com/compute/docs/general-purpose-machines
+  enum Limits {
+    CPUSeries_E2(new TypeLimits(range(2, 33, 2), 512, 8192, false, 0)),
+    CPUSeries_E2MICRO(new TypeLimits(new int[]{}, 1024, 2048, false, 0)),
+    CPUSeries_E2SMALL(new TypeLimits(new int[]{}, 2048, 4096, false, 0)),
+    CPUSeries_E2MEDIUM(new TypeLimits(new int[]{}, 4096, 8192, false, 0)),
+    CPUSeries_N2(
+        new TypeLimits(concat(range(2, 33, 2), range(36, 129, 4)), 512, 8192, true, gbToMb(624))),
+    CPUSeries_N2D(
+        new TypeLimits(new int[]{2, 4, 8, 16, 32, 48, 64, 80, 96}, 512, 8192, true, gbToMb(768))),
+    CPUSeries_N1(
+        new TypeLimits(concat(new int[]{1}, range(2, 97, 2)), 922, 6656, true, gbToMb(624)));
+
+    private final TypeLimits typeLimits;
+
+    Limits(TypeLimits typeLimits) {
+      this.typeLimits = typeLimits;
+    }
+
+    public TypeLimits getTypeLimits() {
+      return typeLimits;
+    }
+  }
+
+  static ImmutableMap<String, Limits> typeLimitsMap = ImmutableMap.<String, Limits>builder()
+      .put("N1", Limits.CPUSeries_N1)
+      .put("N2", Limits.CPUSeries_N2)
+      .put("N2D", Limits.CPUSeries_N2D)
+      .put("E2", Limits.CPUSeries_E2)
+      .put("E2_MICRO", Limits.CPUSeries_E2MICRO)
+      .put("E2_SMALL", Limits.CPUSeries_E2SMALL)
+      .put("E2_MEDIUM", Limits.CPUSeries_E2SMALL)
+      .build();
+
+  static int[] range(int start, int stop, int step) {
+    return IntStream.range(start, stop).filter(x -> (x - start) % step == 0).toArray();
+  }
+
+  static int gbToMb(int value) {
+    return value << 10;
+  }
+
+  static int[] concat(int[] a, int[] b) {
+    int[] result = new int[a.length + b.length];
+    System.arraycopy(a, 0, result, 0, a.length);
+    System.arraycopy(b, 0, result, a.length, b.length);
+    return result;
+  }
```

Comment:
We shouldn't duplicate across samples

Reply:
This helped is crucial to demonstrate in each snippet. Hence, duplicated this across 3 snippets. 
The same procedure has been followed for Python and Go samples as well.

Please advice if this can be made more simpler.